src/main/java/org/openelisglobal/common/rest/provider/PatientSearchRestController.java

```diff
@@ -138,4 +165,110 @@ private PatientSearchWorker getAppropriateWorker(HttpServletRequest request, boo
                 externalID, patientID, guid, dateOfBirth, gender);
     }
 
+    private List<PatientSearchResults> searchPatientInClientRegistry(String lastName, String firstName, String STNumber,
+            String subjectNumber, String nationalID, String patientID, String guid, String dateOfBirth, String gender) {
+        LogEvent.logWarn("PatientSearchRestController", "searchPatientInClientRegistry()",
+                "searchPatientInClientRegistry method has been reached");
+        if (isClientRegistryConfigInvalid()) {
+            return new ArrayList<>();
+        }
+
+        IGenericClient clientRegistry = fhirUtil.getFhirClient(fhirConfig.getClientRegistryServerUrl(),
+                fhirConfig.getClientRegistryUserName(), fhirConfig.getClientRegistryPassword());
+        LogEvent.logWarn("PatientSearchRestController", "searchPatientInClientRegistry()",
+                "ClientRegistry connected successfully");
+
+        IQuery<IBaseBundle> query = buildPatientSearchQuery(clientRegistry, lastName, firstName, STNumber,
+                subjectNumber, nationalID, patientID, guid, dateOfBirth, gender);
+
+        Bundle bundle = query.returnBundle(Bundle.class).execute();
+        List<org.hl7.fhir.r4.model.Patient> externalPatients = new ArrayList<>();
+        if (bundle.hasEntry()) {
+            for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
+                if (entry.getResource() instanceof org.hl7.fhir.r4.model.Patient) {
+                    externalPatients.add((org.hl7.fhir.r4.model.Patient) entry.getResource());
+                    LogEvent.logWarn("PatientSearchRestController", "searchPatientInClientRegistry()",
+                            "externalPatients have been added");
+                }
+            }
+        }
+
+        List<PatientSearchResults> finalResults = new ArrayList<>();
+        for (org.hl7.fhir.r4.model.Patient externalPatient : externalPatients) {
+            // convert fhir object to patient search result
+            // create a rest endpoint to handle import
+            // create a patient search result
+            PatientSearchResults transformedPatientSearchResult = SpringContext.getBean(FhirTransformService.class)
+                    .transformToOpenElisPatientSearchResults(externalPatient);
+            transformedPatientSearchResult.setDataSourceName(MessageUtil.getMessage("patient.cr.source"));
```

Comment:
can you just hard code this data Source Name directly here , since you also hard coded it in the UI ??

Reply:
in case the patient object has no NationalId ,
we can construct a dynamic National ID like "NID-{gender}-{dob}-{iniitials}"